Core/include/Acts/Material/GridSurfaceMaterial.hpp

```diff
@@ -92,52 +93,54 @@ struct IndexedMaterialAccessor {
 template <typename grid_t, typename material_accessor_t = GridMaterialAccessor>
 class GridSurfaceMaterialT : public ISurfaceMaterial {
  public:
+  // Definition of bound (on surface) to grid local representation delegate
+  using BoundToGridLocalDelegate =
+      OwningDelegate<typename grid_t::point_t(const Vector2&),
+                     GridAccess::IBoundToGridLocal>;
+
+  // Definition of global to grid local representation delegate
+  using GlobalToGridLocalDelegate =
+      OwningDelegate<typename grid_t::point_t(const Vector3&),
+                     GridAccess::IGlobalToGridLocal>;
+
   /// Broadcast grid type
   using grid_type = grid_t;
 
   /// Broadcast material accessor type
   using material_accessor_type = material_accessor_t;
 
   /// @brief Constructor for indexed surface material
+  ///
   /// @param grid the index grid steering the access to the material vector
   /// @param materialAccessor the material accessor: from grid, from indexed vector
-  /// @param gcasts global casts -> casts a grid position from global
-  /// @param laccessors local accessors -> accessors to the local grid
-  /// @param transform  transform from global frame into map frame
+  /// @param boundToGridLocal the delegation from bound to grid local frame
+  /// @param globalToGridLocal the delegation from global into grid local frame
   GridSurfaceMaterialT(grid_type&& grid,
                        material_accessor_type&& materialAccessor,
-                       const std::vector<BinningValue>& gcasts,
-                       const std::vector<std::size_t>& laccessors,
-                       const Transform3& transform = Transform3::Identity())
+                       BoundToGridLocalDelegate&& boundToGridLocal,
+                       GlobalToGridLocalDelegate&& globalToGridLocal)
```

Comment:
```suggestion
                       BoundToGridLocalDelegate boundToGridLocal,
                       GlobalToGridLocalDelegate globalToGridLocal)
```

can this be passed by value so the user can decide if it should be moved or not?

Reply:
Absolutely.

Reply:
MMM, the OwningDelegate has (by design) a `unqiue_ptr` to the actual contained/owned class - hence, it looks like I am forced to move, isn't it?

Reply:
Right but the move should be done by the caller I think. It is not strictly necessary to use an rvalue reference here. This is also what we do when we take an `unique_ptr` as an input.

Reply:
this does not work @asalzburger ?

Reply:
Ahh, got it.